engine/access/rpc/backend/backend_scripts.go

```diff
@@ -128,12 +119,42 @@ func (b *backendScripts) executeScriptOnExecutionNode(
 	// *DO NOT* use this hash for any protocol-related or cryptographic functions.
 	insecureScriptHash := md5.Sum(script) //nolint:gosec
 
-	// try each of the execution nodes found
+	// try execution on Archive nodes first
+	if len(b.archiveAddressList) > 0 {
+		startTime := time.Now()
+		for _, rnAddr := range b.archiveAddressList {
+			result, err := b.tryExecuteScriptOnExecutionNode(ctx, rnAddr, blockID, script, arguments)
```

Comment:
should this be?
```suggestion
			result, err := b.tryExecuteScriptOnArchiveNode(ctx, rnAddr, blockID, script, arguments)
```

or does the archive node implement the `execution` protobuf?